File: 240520/최고의 33위치/best-place-of-33.cpp
#include <iostream>
#include <algorithm>
using namespace std;

#define MAX_N 20

int n, ans;
int grid[MAX_N][MAX_N];

bool InRange(int x, int y) {
    return 0<=x && x<n && 0<=y && y<n;
}

void Simulate() {
    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            if(!InRange(i+2, j+2)) continue;

            int cnt = 0;
            for(int k=0; k<3; k++) {
                for(int l=0; l<3; l++) {
                    cnt += grid[i+k][j+l];
                }
            }

            ans = max(ans, cnt);
        }
    }
}

int main() {
    // 여기에 코드를 작성해주세요.
    cin >> n;

    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            cin >> grid[i][j];
        }
    }

    Simulate();

    cout << ans;
    return 0;
}

File: 241104/최고의 33위치/best-place-of-33.py
n = int(input())

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

ans = 0

for i in range(n):
    for j in range(n):
        if i+2 >= n or j+2>= n:
            continue
        cnt = 0
        for a in range(3):
            for b in range(3):
                cnt += grid[i+a][j+b]
        
        ans = max(ans, cnt)

print(ans)